package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -174,132 +167,319 @@ class WaterBridgeAnalysis_OtherFF(WaterBridgeAnalysis):
 
-This table can then be easily turned into, e.g., a `pandas.DataFrame`_, and
-further analyzed::
+        Parameters
+        ----------
+        current : list
+            The current water bridge being analysed is a list of hydrogen bonds from
+            selection 1 to selection 2.
+        output : dict
+            A dictionary where the key is the type of the water bridge and the value
+            is the number of this type of water bridge.
+      '''
+
+      # decompose the first hydrogen bond.
+      s1_index, to_index, (s1_resname, s1_resid, s1_name),
+      (to_resname, to_resid, to_name), dist, angle = current[0]
+      # decompose the last hydrogen bond.
+      from_index, s2_index, (from_resname, from_resid, from_name),
+      (s2_resname, s2_resid, s2_name), dist, angle = current[-1]
+      # if the residue name is ASP and the atom name is OD2 or OD1,
+      # the atom name is changed to OD
+      if s2_resname == 'ASP' and (s2_name == 'OD1' or s2_name == 'OD2'):
+          s2_name = 'OD'
+      # setting up the key which defines this type of water bridge.
+      key = (s1_resname, s1_resid, s1_name, s2_resname, s2_resid, s2_name)
+      # The number of this type of water bridge is incremented by 1.
+      output[key] += 1
+
+  w.count_by_type(analysis_func=analysis)
 
-  import pandas as pd
-  df = pd.DataFrame.from_records(w.table)
+Returns ::
 
+  [(('ARG', 1, 'O', 'ASP', 3, 'OD'), 1.0),]
 
-.. _pandas.DataFrame: http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.html
+Note that the result is arranged in the format of (key, proportion of time). When no
+custom analysis function is supplied, the key is expended for backward compatibility.
+Some people might only interested in contacts between residues and pay no attention
+to the details regarding the atom name. This can also be achieved by supplying an analysis
+function to :meth:`~WaterBridgeAnalysis.count_by_type`.  ::
 
-Classes
--------
+  def analysis(current, output):
+      '''This function defines how the type of water bridge should be specified.
 
-.. autoclass:: WaterBridgeAnalysis
-   :members:
+        Parameters
+        ----------
+        current : list
+            The current water bridge being analysed is a list of hydrogen bonds from
+            selection 1 to selection 2.
+        output : dict
+            A dictionary where the key is the type of the water bridge and the value
+            is the number of this type of water bridge.
+      '''
+
+      s1_index, to_index, (s1_resname, s1_resid, s1_name),
+      (to_resname, to_resid, to_name), dist, angle = current[0]
+      from_index, s2_index, (from_resname, from_resid, from_name),
+      (s2_resname, s2_resid, s2_name), dist, angle = current[-1]
+      # s1_name and s2_name are not included in the key
+      key = (s1_resname, s1_resid, s2_resname, s2_resid)
+      output[key] += 1
+
+  w.count_by_type(analysis_func=analysis)
```

Comment:
`return` missing in example? I'd make `output` optional and create it if `None`. Always return `output`.

Reply:
@orbeckst Sorry the doc is not very compressive. I will write a better doc to explain this. This output is being modified in place.

Reply:
Mention the "in-place" modification requirement for the `analysis()` function.

Reply:
The explanation has been added.